packages/slate-react/src/plugin/react-editor.ts

```diff
@@ -518,6 +518,23 @@ export const ReactEditor = {
       ? anchor
       : ReactEditor.toSlatePoint(editor, [focusNode, focusOffset])
 
+    // If the selection is at the very end of the anchor node, when the anchor and focus are not the same node.
+    if (
+      !isCollapsed &&
+      anchorNode !== focusNode &&
+      anchorNode.nodeValue != null &&
+      anchorNode.nodeValue.length === anchorOffset
+    ) {
+      // Get the next node
+      const newNodeEntry = Editor.next(editor, { at: anchor.path })
```

Comment:
I am not sure this is correct.  Shouldn't you be getting the next `Text` node, in which case I would consider passing a `match: Text.isText`?